allensdk/brain_observatory/behavior/internal/behavior_base.py

```diff
@@ -0,0 +1,130 @@
+import abc
+
+from typing import Dict, NamedTuple
+
+import numpy as np
+import pandas as pd
+
+
+RunningSpeed = NamedTuple("RunningSpeed", [("timestamps", np.ndarray),
+                                           ("values", np.ndarray)])
+
+
+class BehaviorBase(abc.ABC):
+    """Abstract base class implementing required methods for interacting with
+    behavior session data.
+
+    Child classes should be instantiated with a fetch API that implements these
+    methods. Both fetch API and session object should inherit from this base.
+    """
+    @abc.abstractmethod
+    def get_licks(self) -> pd.DataFrame:
+        """Get lick data from pkl file.
+
+        Returns
+        -------
+        np.ndarray
+            A dataframe containing lick timestamps.
+        """
+        raise NotImplementedError()
```

Comment:
Technically you don't need this because it won't let you implement the class without these methods. But I think it's fine to keep.

Reply:
There was a note in the reference docs that abstract method implementations can be called through `super` and I wanted to be explicit and make sure an exception is thrown in that situation.

(https://docs.python.org/3.7/library/abc.html#abc.abstractmethod)